### PR TITLE
Removal of deprecated features for 2.2

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Time to kill more code...  I had this ready for a while :-)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dstansby** [comment]
Maybe needs a rebase to get CircleCI to pass?

**Comment by anntzer** [comment]
just some broken links to removed stuff

File: doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Removal of deprecated features
+``````````````````````````````
+
+The ``matplotlib.finance``, ``mpl_toolkits.exceltools`` and
```
Comment (dstansby) [inline doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst]: Is the finance toolkit now in another package?

File: doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+The ``mpl_toolkits.mplot3d.art3d.iscolor`` function has been removed.
+
+The ``Axes.get_axis_bgcolor``, ``Axes.set_axis_bgcolor``,
+``Bbox.update_from_data``, ``Bbox.update_datalim_numerix``,
+``MaxNLocator.bin_boundaries`` methods have been removed.
+
+The ``bgcolor`` keyword argument to ``Axes`` has been removed.
+
+The ``spectral`` and ``Vega*`` colormaps have been removed.
```
Comment (dstansby) [inline doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst]: `Vega` got renamed right? Might be worth mentioning that here.

File: doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst (hunk 1 above)
Comment (anntzer) [reply doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst]: done

File: doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst (hunk 2 above)
Comment (anntzer) [reply doc/api/api_changes/2017-09-22-AL-removal-of-deprecated-features.rst]: done

**Pull request merged**